### PR TITLE
Chat Translator Plugin - Translates messages to specific language

### PR DESCRIPTION
**Pull request by Farmour**
Translates messages to specified language.
Added google-cloud-translate libraries, wrote plugin functionality for translating public, private, clan and game messages.
Removed plugin setup link, added license header, made DOTALL pattern static

### CODE REVIEW

**Comment by raiyni** [comment]
your pr title should be a summary of what the added feature is and the body providing a better description and/or linking to an issue to close

**Comment by raiyni** [comment]
Same goes for a commit message, description of what/why you changed

File: runelite-client/src/main/java/net/runelite/client/plugins/chattranslate/ChatLanguageMode.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package net.runelite.client.plugins.chattranslate;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/chattranslate/ChatLanguageMode.java]: License header

File: runelite-client/src/main/java/net/runelite/client/plugins/chattranslate/ChatTranslatePlugin.java [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+    /**
+     * Sets up the regex matching for private message username whitelisting
+     */
+    private void updateTranslateFromUsers()
+    {
+        fromUserMatcher = null;
+        if (config.translateFromUsers().trim().equals("*") && config.translatePrivateMessages())
+        {
+            fromUserMatcher = Pattern.compile(".", Pattern.DOTALL);
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/chattranslate/ChatTranslatePlugin.java]: Compite patterns at the top of the class in a `final static`

**Comment by alexgermann** [comment]
Is this plugin ever going to be merged?

**Comment by raiyni** [comment]
Polywoof translation plugin on the plugin hub does this.

**Pull request closed without merging**